pkgs/by-name/po/polyglot/package.nix

```
@@ -0,0 +1,56 @@
+{
+  lib,
+  fetchFromGitHub,
+  jre,
+  makeWrapper,
+  maven,
+  libGL,
+  xdg-utils,
+  zip,
+  zlib
+}:
+maven.buildMavenPackage rec {
+  pname = "polyglot";
+  version = "3.5.1";
+  src = fetchFromGitHub {
+    owner = "DraqueT";
+    repo = "PolyGlot";
+    rev = version;
+    hash = "sha256-E7wLhohOpWGzXe1zEO9a8aFIVT7/34Wr0dsRzpuf+eY=";
+  };
+
+  preBuild = ''
+    echo "${version}" > assets/assets/org/DarisaDesigns/version
+    cd docs
+    zip -r ../assets/assets/org/DarisaDesigns/readme *
+    cd ../packaging_files/example_lexicons
+    zip -r ../../assets/assets/org/DarisaDesigns/exlex *
+    cd ../..
+  '';
+
+  mvnHash = "sha256-u30LOaDA/AQO0bTTLxpDJsbiVeLd/nAvyWsDPXp2eb0=";
+  mvnParameters = "-DskipTests";
+
+  nativeBuildInputs = [ makeWrapper zip ];
+  runtimeDeps = [ libGL xdg-utils zlib ];
+  installPhase = ''
+    mkdir -p $out/bin $out/share/PolyGlotLinA
+    install -Dm644 --target-directory=$out/share/PolyGlotLinA target/PolyGlotLinA-${version}-jar-with-dependencies.jar
+
+    makeWrapper ${jre}/bin/java $out/bin/PolyGlot \
+      --add-flags "-Djpackage.app-version=${version}" \
+      --add-flags "-jar $out/share/PolyGlotLinA/PolyGlotLinA-${version}-jar-with-dependencies.jar"
+  '';
+
+  meta = {
+    description = "A conlang construction toolkit";
+    homepage = "https://draquet.github.io/PolyGlot/readme.html";
+    changelog = "https://github.com/DraqueT/PolyGlot/releases/tag/${version}";
+    license = lib.licenses.mit;
+    maintainers = [ lib.maintainers.noodlez1232 ];
+    platforms = lib.platforms.linux;
```

The PolyGlot website has binaries for macOS too
```suggestion
    platforms = lib.platforms.all;
```

---

I don't have a Mac, and the build process for Mac is different, so I'm not sure if I could figure out how to build it. If you have any advice on that, I'd appreciate it.

---

I decided to just push it with it set to `.all` and we'll see. I'll debug using CI.